tldw_Server_API/app/api/v1/endpoints/billing_webhooks.py

```
@@ -0,0 +1,117 @@
+            status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
+            detail="Stripe is not configured",
+        )
+
+    try:
+        event = stripe_client.construct_webhook_event(body, stripe_signature)
+    except ValueError as e:
+        logger.warning(f"Invalid Stripe webhook signature: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Invalid webhook signature",
+        )
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)